src/main/java/javaslang/control/Validation.java

```diff
@@ -0,0 +1,253 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+import javaslang.Function1;
+import javaslang.algebra.Functor;
+import javaslang.collection.List;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * An implementation similar to scalaz's <a href="http://eed3si9n.com/learning-scalaz/Validation.html">Validation</a> control.
+ *
+ * <p>
+ * The Validation type is different from a Monad type, it is an applicative
+ * functor. Whereas a Monad will short circuit after the first error, the
+ * applicative functor will continue on, accumulating ALL errors. This is
+ * especially helpful in cases such as validation, where you want to know
+ * all the validation errors that have occurred, not just the first one.
+ * </p>
+ *
+ * <pre>
+ * <code>
+ * Validation.Success:
+ * Validation&lt;List&lt;String&gt;,Integer&gt; valS = Validation.success(5);
+ *
+ * Validation.Failure:
+ * Validation&lt;List&lt;String&gt;,Integer&gt; valE = Validation.failure(List.of("error1","error2"));
+ * </code>
+ * </pre>
+ *
+ * @param <E> Value type in the case of failure.
+ * @param <T> Value type in the case of success.
+ * @author Eric Nelson
+ * @since 2.0.1
+ * @see <a href="http://eed3si9n.com/learning-scalaz/Validation.html">Validation</a>
+ */
+public interface Validation<E,T> extends Functor<T> {
+
+    static <E,T> Validation<E,T> success(T value) {
+        return new Success<>(value);
+    }
+
+    static <E,T> Validation<E,T> success(Supplier<T> supplier) {
+        return new Success<>(supplier.get());
+    }
+
+    static <E,T> Validation<E,T> failure(E error) {
+        return new Failure<>(error);
+    }
+
+    static <E,T> Validation<E,T> failure(Supplier<E> supplier) {
+        return new Failure<>(supplier.get());
+    }
+
+    boolean isSuccess();
+
+    boolean isFailure();
+
+    Object get();
+
+    void foreach(Consumer<T> f);
```

Comment:
We use `forEach` instead of `foreach`. We will get it automatically when implementing `Value`. However, before implementing common Javaslang interfaces we should create a working _spike_ in order to keep the complexity low. We don't care for now.